eclair-core/src/main/scala/fr/acinq/eclair/channel/fsm/Channel.scala

```diff
@@ -350,39 +350,63 @@ class Channel(val nodeParams: NodeParams, val channelKeys: ChannelKeys, val wall
           goto(CLOSED) using closing
         case closing: DATA_CLOSING =>
           val localPaysClosingFees = closing.commitments.params.localParams.paysClosingFees
-          // we don't put back the WatchSpent if the commitment tx has already been published and the spending tx already reached mindepth
           val closingType_opt = Closing.isClosingTypeAlreadyKnown(closing)
           log.info(s"channel is closing (closingType=${closingType_opt.map(c => EventType.Closed(c).label).getOrElse("UnknownYet")})")
-          // if the closing type is known:
-          // - there is no need to watch the funding tx because it has already been spent and the spending tx has already reached mindepth
-          // - there is no need to attempt to publish transactions for other type of closes
-          // - there is a single commitment, the others have all been invalidated
+          // If the closing type is known:
+          //  - there is no need to attempt to publish transactions for other type of closes
+          //  - there may be 3rd-stage transactions to publish
+          //  - there is a single commitment, the others have all been invalidated
+          val commitment = closing.commitments.latest
           closingType_opt match {
             case Some(c: Closing.MutualClose) =>
               doPublish(c.tx, localPaysClosingFees)
             case Some(c: Closing.LocalClose) =>
-              val secondStageTransactions = Closing.LocalClose.SecondStageTransactions(c.localCommitPublished.claimMainDelayedOutputTx, c.localCommitPublished.claimAnchorTx_opt, c.localCommitPublished.htlcTxs.values.flatten.toSeq)
-              doPublish(c.localCommitPublished, secondStageTransactions, closing.commitments.latest)
-              val thirdStageTransactions = Closing.LocalClose.ThirdStageTransactions(c.localCommitPublished.claimHtlcDelayedTxs)
+              val (_, secondStageTransactions) = Closing.LocalClose.claimCommitTxOutputs(channelKeys, commitment, c.localCommitPublished.commitTx, nodeParams.currentBitcoinCoreFeerates, nodeParams.onChainFeeConf, closing.finalScriptPubKey)
```

Comment:
Here and in the other case matches we re-compute 2nd and 3rd stage transactions. Does it mean that, if we restart and the previously published transactions have not confirmed yet, we will automatically publish new (possibly more expensive) transactions ?